src/main/scala/cognite/spark/v1/FlexibleDataModelRelationUtils.scala

```diff
@@ -668,58 +665,81 @@ object FlexibleDataModelRelationUtils {
       case Success(i) => get(i).map(FieldSpecified(_))
     }
 
+  private def getListPropAsSeq[T](propertyName: String, row: Row, index: Integer): Seq[T] =
+    Try(row.getSeq[T](index)) match {
+      case Success(x) => x
+      case Failure(_) =>
+        Try(row.getAs[Array[T]](index).toSeq) match {
+          case Success(x) => x
+          case Failure(err) =>
+            throw new CdfSparkException(
+              f"""Could not deserialize property $propertyName as an array of the expected type""",
+              err)
+        }
+    }
+
   // scalastyle:off cyclomatic.complexity method.length
   private def toInstancePropertyValueOfList(
       row: Row,
       schema: StructType,
       propertyName: String,
-      propDef: CorePropertyDefinition): Either[Throwable, OptionalField[InstancePropertyValue]] =
+      propDef: CorePropertyDefinition,
+      instanceSpace: Option[String]
+  ): Either[Throwable, OptionalField[InstancePropertyValue]] =
     lookupFieldInRow(row, schema, propertyName, propDef.nullable.getOrElse(true)) { i =>
       propDef.`type` match {
-        case p: TextProperty if p.isList =>
-          val strSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-          Try(InstancePropertyValue.StringList(skipNulls(strSeq).map(String.valueOf))).toEither
-        case p @ PrimitiveProperty(PrimitivePropType.Boolean, _) if p.isList =>
-          val boolSeq = Try(row.getSeq[Boolean](i)).getOrElse(row.getAs[Array[Boolean]](i).toSeq)
-          Try(InstancePropertyValue.BooleanList(boolSeq)).toEither
-        case p @ PrimitiveProperty(PrimitivePropType.Float32, _) if p.isList =>
-          val floatSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-          tryAsFloatSeq(floatSeq, propertyName)
+        case _: DirectNodeRelationProperty =>
+          val structSeq = getListPropAsSeq[Row](propertyName, row, i)
+          tryAsDirectNodeRelationList(structSeq, propertyName, instanceSpace)
+            .map(InstancePropertyValue.ViewDirectNodeRelationList)
+        case _: TextProperty =>
+          Try({
+            val strSeq = getListPropAsSeq[String](propertyName, row, i)
+            InstancePropertyValue.StringList(skipNulls(strSeq))
+          }).toEither
+        case _ @PrimitiveProperty(PrimitivePropType.Boolean, _) =>
+          Try({
+            val boolSeq = getListPropAsSeq[Boolean](propertyName, row, i)
+            InstancePropertyValue.BooleanList(boolSeq)
+          }).toEither
+        case _ @PrimitiveProperty(PrimitivePropType.Float32, _) =>
+          val floatSeq = getListPropAsSeq[Any](propertyName, row, i)
+          tryConvertNumberSeq(floatSeq, propertyName, Float.toString, safeConvertToFloat)
```

Comment:
```suggestion
          tryConvertNumberSeq(floatSeq, propertyName, "Float", safeConvertToFloat)
```
Multiple possibilities for error here otherwise
- `Float.toString` does look like a lambda `(x: Float) => x.toString` but it isn't the inention here afaics
- intellij says it'd rather be 
```scala
  /** The String representation of the scala.Float companion object. */
  override def toString = "object scala.Float"
```
which isn't the user-facing type to show (still readable but with more noise than before)